python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py

```diff
@@ -1,200 +1,265 @@
-from dataclasses import dataclass, field
 from datetime import datetime
-from enum import Enum, auto
-from typing import Any
+from enum import Enum
+from typing import Optional
+
+from pydantic import BaseModel, Field
 
 DATAHUB_DATE_FORMAT = "%Y%m%d"
 
 
-@dataclass
-class CatalogueMetadata:
-    name: str
-    description: str
-    owner: str
-    tags: list[str] = field(default_factory=list)
+class DatabaseStatus(Enum):
+    PROD = "PROD"
+    PREPROD = "PREPROD"
+    DEV = "DEV"
+
 
+class RelationshipType(Enum):
+    PARENT = "PARENT"
+    PLATFORM = "PLATFORM"
 
-@dataclass
-class DataLocation:
+
+# Needs refining
+class SecurityClassification(Enum):
     """
-    A representation of where the data can be found
-    (in our case, glue/athena)
+    Government security classification
     """
 
-    fully_qualified_name: str
-    platform_type: str = "glue"
-    platform_id: str = "glue"
+    OFFICIAL = "OFFICIAL"
 
 
-class DataProductStatus(Enum):
-    DRAFT = auto()
-    PUBLISHED = auto()
-    RETIRED = auto()
+class EntityRef(BaseModel):
+    """
+    A reference to another entity in the metadata graph.
+    """
 
+    id: str = Field(description="The identifier of the entity being linked to.")
```

Comment:
is the `id` always a urn?
```suggestion
    id: str = Field(description="The urn identifier of the entity being linked to.")
```

Reply:
I believe so. The only reason we're using `id` at all is because originally, the models were independent of catalogue.

We could do a search and replace or id -> urn if nobody objects.